managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -116,6 +116,22 @@ public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
                 recycle();
                 return;
             }
+
+            int toAckEntryNum = 0;
+            List<Entry> skippedEntries = new ArrayList<>();
+            PositionImpl startPosition = readPosition;
+            PositionImpl endPosition = (PositionImpl) nexReadPosition;
+            while (startPosition.compareTo(endPosition) < 0) {
+                skippedEntries.add(EntryImpl.createSkippedEntry(startPosition.ledgerId, startPosition.entryId));
+                startPosition = ledger.getNextValidPosition(startPosition);
+                toAckEntryNum++;
+                if (toAckEntryNum > cursor.getConfig().getMaxAckEntryNumForAutoSkipNonRecoverableData()) {
```

Comment:
If the state of the cursor is like this:

```
read position {1:0}
individual deleted messages [ {1:0}, {1:50001} ]
```

After the entry filter, the `nexReadPosition` will be {1,10000} and  `filteredEntries` will be {1,0}, then `maxAckEntryNumForAutoSkipNonRecoverableData` could not work perfect. Can we let `cursor` do the filtering?

Reply:
I will fixed